dotnet/src/SemanticKernel.Abstractions/PromptTemplate/PromptTemplateConfig.cs

```diff
@@ -169,7 +169,29 @@ internal IReadOnlyList<KernelParameterMetadata> GetKernelParametersMetadata()
     /// <exception cref="ArgumentException">Thrown when the deserialization returns null.</exception>
     public static PromptTemplateConfig FromJson(string json)
     {
-        var result = JsonSerializer.Deserialize<PromptTemplateConfig>(json, JsonOptionsCache.ReadPermissive);
-        return result ?? throw new ArgumentException("Unable to deserialize prompt template config from argument. The deserialization returned null.", nameof(json));
+        var promptTemplateConfig = JsonSerializer.Deserialize<PromptTemplateConfig>(json, JsonOptionsCache.ReadPermissive);
+        if (promptTemplateConfig == null)
+        {
+            throw new ArgumentException("Unable to deserialize prompt template config from argument. The deserialization returned null.", nameof(json));
+        }
+
+        // Prevent the default value from being any type other than a string.
```

Comment:
Why?

Reply:
It's a temporary limitation that allows us to shape the public API surface (changing the type of the `Default` property to object) now, before the release. This helps avoid a breaking change while we work on a proper solution for dealing with the different deserialization outputs of JSON/YAML prompt configurations.

Reply:
Ok. It'd be good if the comment indicated the why.